File: API/Features/Reservations/Dtos/Read/ReservationReadDto.cs
using System.Collections.Generic;
using API.Infrastructure.Interfaces;

namespace API.Features.Reservations {

    public class ReservationReadDto : IMetadata {

        public string ReservationId { get; set; }
        public ReservationBoatReadDto Boat { get; set; }
        public ReservationCaptainReadDto Captain { get; set; }
        public List<ReservationBerthReadDto> Berths { get; set; }
        public string FromDate { get; set; }
        public string ToDate { get; set; }
        public int Days { get; set; }
        public bool IsDocked { get; set; }
        public bool IsDryDock { get; set; }
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}

File: FrontEnd/src/app/features/reservations/classes/dtos/read/reservation-read-dto.ts
import { Metadata } from 'src/app/shared/classes/metadata'
import { ReservationBerthReadDto } from './reservation-berth-read-dto'
import { ReservationBoatReadDto } from './reservation-boat-read-dto'

export interface ReservationReadDto extends Metadata {

    reservationId: string
    boat: ReservationBoatReadDto
    berths: ReservationBerthReadDto[]
    fromDate: string
    toDate: string
    days: number
    isPassingBy: boolean
    isDocked: false,
    isDryDock: false,
    isDeleted: false,
    postAt: string
    postUser: string
    putAt: string
    putUser: string

}
